vsphere/resource_vsphere_guest_os_customization_test.go

```diff
@@ -0,0 +1,176 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGOSCWindowsAllPropsWorkGroup(goscResourceName, goscName),
+				Check:  testAccGOSCExists(goscResourceName, goscName, true),
+			},
+		},
+	})
+}
+
+func TestAccResourceVSpherGOSC_linux(t *testing.T) {
+	goscName := acctest.RandomWithPrefix("lin")
+	goscResourceName := acctest.RandomWithPrefix("gosc")
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			RunSweepers()
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGOSCExists(goscResourceName, goscName, false),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGOSCLinux(goscResourceName, goscName),
+				Check:  testAccGOSCExists(goscResourceName, goscName, true),
+			},
+		},
+	})
+}
+
+func TestAccResourceVSpherGOSC_sysprep(t *testing.T) {
+	goscName := acctest.RandomWithPrefix("lin")
+	goscResourceName := acctest.RandomWithPrefix("gosc")
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			RunSweepers()
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccGOSCExists(goscResourceName, goscName, false),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGOSCWindowsPrep(goscResourceName, goscName),
+				Check:  testAccGOSCExists(goscResourceName, goscName, true),
+			},
+		},
+	})
+}
+
+func testAccGOSCExists(resourceName string, goscName string, expectToExist bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		resource := fmt.Sprintf("vsphere_guest_os_customization.%s", resourceName)
+		vars, err := testClientVariablesForResource(s, resource)
+		if err != nil {
+			return err
+		}
+		_, err = guestoscustomizations.FromName(vars.client, goscName)
+		if err != nil && expectToExist {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccGOSCWindows(resourceName string, goscName string) string {
+	return fmt.Sprintf(`
+		resource "vsphere_guest_os_customization" %q {
+			name = %q
+			type = "Windows"
+			spec {
+				windows_options {
+					computer_name = "Windows-Computer-Name"
+				}
+			}
+		}
+	`,
+		resourceName,
+		goscName,
+	)
+}
+
+func testAccGOSCWindowsAllPropsWorkGroup(resourceName string, goscName string) string {
+	return fmt.Sprintf(`
+		resource "vsphere_guest_os_customization" %q {
+			name = %q
+			type = "Windows"
+			spec {
+				windows_options {
+					run_once_command_list = ["command-1", "command-2"]
+					computer_name = "Windows-Computer-Name"
+					auto_logon = false
+					auto_logon_count = 0
+					admin_password = "Pass-123"
+					time_zone = 125 #(UTC+02:00) Helsinki, Kyiv, Riga, Sofia, Tallinn, Vilnius
+					workgroup = "Windows-Workgroup"
+				}
+			}
+		}
+	`,
+		resourceName,
+		goscName,
+	)
+}
+
+func testAccGOSCWindowsPrep(resourceName string, goscName string) string {
+	return fmt.Sprintf(`
+		resource "vsphere_guest_os_customization" %q {
+			name = %q
+			type = "Windows"
+			spec {
+				windows_sysprep_text = "Test sysprep text"
```

Comment:
```suggestion
				windows_sysprep_text = "${file("${path.module}/sysprep.xml")}"
```

Reply:
Tried to do this, but it looks like the sysprep.xml file can not be accessed by the HCL in the acceptance test, no mather where I place it, so I will drop this one. The dummy text is good enough for testing purposes IMHO. I will appreciate a hint on this one